spec/cucumber/core/report/summary_spec.rb

```diff
@@ -61,14 +61,106 @@ module Report
             expect(@summary.test_cases.total).to eq(1)
           end
 
-          it 'handles flaky test cases' do
-            allow(test_case).to receive(:==).and_return(false, true)
-            event_bus.send(:test_case_finished, test_case, failed_result)
-            event_bus.send(:test_case_finished, test_case, passed_result)
+          context 'handles flaky test cases' do
```

Comment:
If possible it would be better to write each of these as just 1 level of nesting under context with a better description. It's a bit hard for me to work out what's different here. Admittedly I don't know the code that well.